src/core/ext/filters/client_channel/http_proxy.cc

```diff
@@ -53,6 +55,33 @@
 namespace grpc_core {
 namespace {
 
+static bool serverInCIDRRange(std::string server_host, absl::string_view no_proxy_entry) {
+  auto server_address = StringToSockaddr(server_host, 0);
+  if (!server_address.ok()) {
+    return false;
+  }
+
+  std::vector<absl::string_view> cidr = absl::StrSplit(no_proxy_entry, '/', absl::SkipEmpty());
+  if (cidr.size() == 0 || cidr.size() > 2) {
+    return false;
+  }
+
+  auto proxy_address = StringToSockaddr(cidr[0], 0);
```

Comment:
We may want to parse port numbers for exact IP matches. Golang does this: https://github.com/golang/go/blob/f771edd7f92a47c276d65fbd9619e16a786c6746/src/vendor/golang.org/x/net/http/httpproxy/proxy.go#L38-L50